test/LiquidDemocracy.test.js

```diff
@@ -101,9 +101,71 @@ contract('LiquidDemocracy', function (accounts){
             representative1.should.equal(accounts[1]);
             representative2.should.equal(accounts[1]);
             representativeWeight.toNumber().should.equal(3);
-            representee1Weight.toNumber().should.equal(0);
-            representee2Weight.toNumber().should.equal(0);
+            representee1Weight.toNumber().should.equal(1);
+            representee2Weight.toNumber().should.equal(1);
         });
 
     });
+
+    describe('revokes produce the expected output', function () {
+
+        it('delegating and revoking are inverse functions', async function () {
+            const representee = accounts[0];
+
+            await this.liquidDemocracyBallot.delegate(accounts[1], { from: representee });
+            await this.liquidDemocracyBallot.revoke({ from: representee });
+
+            const newRepresentative = await this.liquidDemocracyBallot.getMyRepresentative({ from: representee })
+            const newRepresentativeWeight = await this.liquidDemocracyBallot.getMyWeight({ from: accounts[1] })
+            const newRepresenteeWeight = await this.liquidDemocracyBallot.getMyWeight({ from: representee })
+
+            newRepresentative.should.equal(representee);
+            newRepresentativeWeight.toNumber().should.equal(1);
+            newRepresenteeWeight.toNumber().should.equal(1);
+        });
+
+        it('revoking on the leaf of transitive delegation gives the expected results', async function () {
+            const representee = accounts[0];
+            const representative1 = accounts[1];
+            const representative2 = accounts[2];
+
+            await this.liquidDemocracyBallot.delegate(representative1, { from: representee });
+            await this.liquidDemocracyBallot.delegate(accounts[2], { from: representative1 });
```

Comment:
Should we use the already declared variable `representative2` here as first argument?